locator/tablets.cc

```diff
@@ -430,6 +430,42 @@ class tablet_effective_replication_map : public effective_replication_map {
         return result;
     }
 
+    std::optional<tablet_routing_info> check_locality(const token& search_token) const override {
+        auto&& tablets = get_tablet_map();
+        auto tid = tablets.get_tablet_id(search_token);
+        auto&& info = tablets.get_tablet_info(tid);
+        auto host = get_token_metadata().get_my_id();
+        auto shard = this_shard_id();
+
+        auto make_tablet_routing_info = [&] {
+            dht::token first_token;
+            if (tid == tablets.first_tablet()) {
+                first_token = dht::first_token();
```

Comment:
Since the range is (first_token, last_token], this will incorrectly exclude keys which are mapped to `dht::first_token()`, no?

We can use dht::minimum_token(), which is not mapped to any key.